vertx-web/src/main/java/io/vertx/ext/web/sstore/impl/SessionImpl.java

```diff
@@ -74,6 +75,18 @@ public String id() {
   }
 
   @Override
+  public Session regenerateId() {
+    if (oldId == null) {
+      // keep track of just the first one since the
+      // regeneration during the remaining lifecycle are ephemeral
+      oldId = id;
+    }
+    id = UUID.randomUUID().toString();
```

Comment:
I'd be careful with this code with JDK8, in JDK8 there were some changes to the secure random code, for instance there are 4 providers (on unices):

* SHA1PRNG (Initial seeding is currently done via a combination of system attributes and the java.security entropy gathering device)
* NativePRNG (nextBytes() uses /dev/urandom, generateSeed() uses /dev/random)
* NativePRNGBlocking (nextBytes() and generateSeed() use /dev/random)
* NativePRNGNonBlocking (nextBytes() and generateSeed() use /dev/urandom)

For portability reasons we shouldn't pick any `NativePRNG*` since they won't be available on Windows where there is another provider 

* Windows-PRNG (nextBytes() uses Windows CryptGenRandom() API)

When instantiating a secure random (on unices) without preference the default choice is a _mixed_ mode:

* /dev/urandom (non-blocking) for nextBytes()
* /dev/random (blocking) for generateSeed()

Which if you're running on Windows won't be present and you'll end up with `SHA1PRNG`.